src/app/services/data/index.ts

```diff
@@ -55,13 +67,30 @@ export const MeshService = {
     const labels = item.labels ?? {}
     const name = labels['kuma.io/display-name'] ?? item.name
     const namespace = labels['k8s.kuma.io/namespace'] ?? ''
-
     return {
       ...item,
-      id: item.name,
       config: item,
+      id: item.name,
       name,
       namespace,
+      labels,
+      spec: ((item = {}) => {
+        return {
+          ports: Array.isArray(item.ports) ? item.ports : [],
+          selector: ((item = {}) => {
+            return {
+              dataplaneTags: Object.keys(item.dataplaneTags ?? {}).length > 0 ? item.dataplaneTags! : {},
+            }
+          })(item.selector),
+        }
+      })(item.spec),
+      status: ((item = {}) => {
+        return {
+          tls: typeof item.tls !== 'undefined' ? item.tls : { status: '' },
+          vips: Array.isArray(item.vips) ? item.vips : [],
+          addresses: Array.isArray(item.addresses) ? item.addresses : [],
+        }
+      })(item.status),
```

Comment:
Lots of handwritten code here just to make sure things are set 🤔 , if we can't sort this some other way, it would be good longer term to see if we can automatically generate this sort of code, or at least stick to using a pattern that is common and easy to read and copy/paste throughout the application.